### PR TITLE
Various minor changes

### PR DESCRIPTION
**Pull request by KamasamaK**
### Breaking changes

- Removed `RenameOptions.id` as it was already deprecated and never specified for `StaticRegistrationOptions`
- Removed `SemanticTokenTypes.Member` as it was already deprecated and is not specified
- Removed `TraceValue.Message` as it was already deprecated and is not specified
- Changed `TraceValue` to be `final` matching similar classes
- Removed duplicate `ResponseErrorCode` as it has been deprecated for several versions
- Removed `ResponseErrorCode.serverErrorStart` and `ResponseErrorCode.serverErrorEnd` as they were already deprecated and just boundaries not actual error codes

### Other changes

- Improve many docblocks
- Move many classes with only static members into `Protocol.xtend`
- Improve JSON-RPC Markdown document

### CODE REVIEW

**Comment by jonahgraham** [comment]
@KamasamaK thanks for doing this cleanup. I have only reviewed as far as the changelog and that was fine by me.

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/services/LanguageServer.java [hunk 1]
```diff
@@ -30,17 +30,19 @@ public interface LanguageServer {
...
-	 *  - notifications should be dropped, except for the exit notification.
+	 * <li>notifications should be dropped, except for the exit notification.
 	 *    This will allow the client to exit a server without an initialize request.
-	 *  
+	 * </ul>
+	 * <p>
 	 * Until the server has responded to the initialize request with an InitializeResult,
 	 * the client must not send any additional requests or notifications to the server.
-	 * 
+	 * <p>
```
Comment (cdietrich) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/services/LanguageServer.java:45]: there are these closed?

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/services/LanguageServer.java [hunk 2]
```diff
@@ -30,17 +30,19 @@ public interface LanguageServer {
...
 	 * If the server receives requests or notifications before the initialize request, it should act as follows:
-	 *  - for a request, the response should be errored with:
-	 *    {@link org.eclipse.lsp4j.jsonrpc.messages.ResponseErrorCode#serverNotInitialized}.
+	 * <p><ul>
+	 * <li>for a request, the response should be errored with:
+	 *    {@link org.eclipse.lsp4j.jsonrpc.messages.ResponseErrorCode#ServerNotInitialized}.
 	 *    The message can be picked by the server.
-	 *  - notifications should be dropped, except for the exit notification.
+	 * <li>notifications should be dropped, except for the exit notification.
```
Comment (cdietrich) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/services/LanguageServer.java:39]: where is this one closed?

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/SymbolTag.java [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 /**
  * Symbol tags are extra annotations that tweak the rendering of a symbol.
- * 
+ * <p>
```
Comment (cdietrich) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/SymbolTag.java:17]: where is this one closed

**Comment by cdietrich** [comment]
javadoc changes look suspicious. tags dont seem to be closed

**Comment by pisv** [comment]
Just a note that it does not seem required to close these tags in Javadoc, as examples in https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html show. Not closing these tags also matches common practice in other parts of LSP4J code base, including `Protocol.xtend`.

**Comment by cdietrich** [comment]
understood

**Review by cdietrich — APPROVED** [review]

**Pull request merged**